### DISCUSSION TITLE
Looking for additional info on hybrid training with bootstrap

### DISCUSSION THREAD

**Discussion by ZoltanV-V** [lava-nc/lava-dl#166] [Q&A]
Hi! I've been attempting to train a model on a custom dataset with 10 classes. I've defined the model using bootstrap and have tried playing around with the training parameters to see if I can improve the performance of the SNN but have been unsuccessful. Looking at the accuracy.png plot generated after training and validation, the ANN has testing accuracy in the mid 90's but the SNN is only in low to mid 60s. I read about the hybridization training possibly improving this gap but it isn't shown in the mnist.py tutorial in bootstrap. Is there a way I can get more info on how to utilize this feature? Thank you so much!

**Comment by bamsumit** [comment]
When you create your [scheduler](https://lava-nc.org/lava-lib-dl/bootstrap/routine.html#lava.lib.dl.bootstrap.routine.Scheduler), you can pass a list of crossover parameters like this 
```python
scheduler = bootstrap.routine.Scheduler(num_sample_iter=10, sample_period=10, crossover_epochs=[10, 20, 30, 40, 50])
```
This means
* after epoch 10, block 0 will always run on SNN mode.
* after epoch 20, block 0 & 1 will always run on SNN mode.
* after epoch 30, block 0, 1 & 2 will always run on SNN mode.
* and so on

I do suspect your neurons are saturating (high firing rate) which is the cause of the accuracy difference. It would be useful to look at the neuron firing rate or even the input range. Its good to normalize the inputs between -1 and 1 or 0 and 1 as needed.

**Reply by ZoltanV-V** [reply]
Thank you so much, I'll give this a try and see if I am able to improve the performance!
